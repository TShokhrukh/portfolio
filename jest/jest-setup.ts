import '@testing-library/jest-dom'
import { TextEncoder, TextDecoder } from 'util'
import { IntersectionObserverMock } from './mock/IntersectionObserverMock '

// @ts-ignore
global.TextEncoder = TextEncoder
// @ts-ignore
global.TextDecoder = TextDecoder
// @ts-ignore
global.IntersectionObserver = IntersectionObserverMock
