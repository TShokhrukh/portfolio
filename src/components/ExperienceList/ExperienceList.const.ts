import { cn } from '@bem-react/classname'

export const cnExperienceList = cn('ExperienceList')
