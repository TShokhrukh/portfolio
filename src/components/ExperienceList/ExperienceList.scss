.ExperienceList {
  --experience-list-gap: #{$spacer * 2};
  --experience-list-width: 100%;

  display: flex; 
  flex-direction: column;
  gap: var(--experience-list-gap);
}

@include media-breakpoint-up(xl) {
  .ExperienceList {
    --experience-list-width: 25%;
  
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }

  .ExperienceList-Item {
    width: var(--experience-list-width);
  }
}