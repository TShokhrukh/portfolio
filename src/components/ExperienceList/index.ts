export * from './build'
export * from './__types__'
export { ExperienceList as ExperienceListBase } from './ExperienceList'
