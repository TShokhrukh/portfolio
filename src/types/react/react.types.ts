import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
