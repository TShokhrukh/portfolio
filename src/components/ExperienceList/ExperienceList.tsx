import React, { FC } from 'react'
import { ExperienceCard } from '~/components/ExperienceCard/ExperienceCard'
import { cnExperienceList } from './ExperienceList.const'

import type { IExperienceListProps } from './__types__'

import './ExperienceList.scss'

export const ExperienceList: FC<IExperienceListProps> = ({
  component: Component = 'div',
  className,
  experience,
  ...props
}) => {
  return (
    <Component {...props} className={cnExperienceList({}, [className])}>
      {experience.map(({ id, ...e }) =>
        <ExperienceCard key={id} experience={e} className={cnExperienceList('Item', [])} />
      )}
    </Component>
  )
}
