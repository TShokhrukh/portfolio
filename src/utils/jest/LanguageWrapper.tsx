import React, { ReactNode } from 'react'
import { TranslationsProvider } from '@eo-locale/react'

import type { Locale } from '@eo-locale/core'

interface IWrapperProps {
  children: ReactNode,
  language?: string,
  locales?: Locale[]
}

const defLocales: Locale[] = [
  {
    language: 'en',
    messages: {}
  }
]

export const LanguageWrapper: React.FC<IWrapperProps> = ({
  children,
  language = 'en',
  locales = defLocales
}) => {
  return (
    <TranslationsProvider language={language} locales={locales} onError={() => null}>
      {children}
    </TranslationsProvider>
  )
}
