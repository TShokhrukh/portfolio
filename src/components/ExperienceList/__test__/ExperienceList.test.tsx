/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { LanguageWrapper } from '~/utils/jest/LanguageWrapper'
import { ExperienceList } from '../ExperienceList'

describe('ExperienceList', () => {
  test('must return the component template', () => {
    const experience = [
      {
        id: '1',
        icon: 'home',
        locales: {
          en: {
            body: 'home_body',
            skill: 'home_skill'
          }
        }
      },
      {
        id: '2',
        icon: 'user',
        locales: {
          en: {
            body: 'user_body',
            skill: 'user_skill'
          }
        }
      }
    ]
    render(
      <LanguageWrapper>
        <ExperienceList experience={experience} data-testid='experience_list' />
      </LanguageWrapper>
    )

    expect(screen.getByTestId('experience_list')).toMatchSnapshot()
  })
})
