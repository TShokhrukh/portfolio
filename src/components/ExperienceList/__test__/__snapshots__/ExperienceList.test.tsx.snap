// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceList must return the component template 1`] = `
<div
  class="ExperienceList"
  data-testid="experience_list"
>
  <div
    class="Card ExperienceCard ExperienceList-Item"
  >
    <span
      aria-hidden="true"
      class="fa fa-home Icon ExperienceCard-Icon"
    />
    <p
      class="ExperienceCard-Body"
    >
      home_body
    </p>
    <h3
      class="ExperienceCard-Skill"
    >
      home_skill
    </h3>
  </div>
  <div
    class="Card ExperienceCard ExperienceList-Item"
  >
    <span
      aria-hidden="true"
      class="fa fa-user Icon ExperienceCard-Icon"
    />
    <p
      class="ExperienceCard-Body"
    >
      user_body
    </p>
    <h3
      class="ExperienceCard-Skill"
    >
      user_skill
    </h3>
  </div>
</div>
`;
